python/graphstorm/gconstruct/transform.py

```diff
@@ -1080,23 +1100,55 @@ def process_features(data, ops):
         The data stored as a dict.
     ops : list of FeatTransform
         The operations that transform features.
+    ext_mem: str or None
+        The address of external memory
 
     Returns
     -------
     dict : the key is the data name, the value is the processed data.
     """
     new_data = {}
     for op in ops:
-        res = op(data[op.col_name])
-        assert isinstance(res, dict)
-        for key, val in res.items():
-            # Check if has 1D features. If yes, convert to 2D features
-            if len(val.shape) == 1:
-                if isinstance(val, ExtMemArrayWrapper):
-                    val = val.to_numpy().reshape(-1, 1)
+        hash_hex_feature_path = generate_hash()
+        feature_path = 'feature_{}_{}'.format(op.feat_name, hash_hex_feature_path)
+        feature_path = ext_mem + feature_path if ext_mem is not None \
+            else feature_path
+        if isinstance(op.col_name, str):
+            col_name = [op.col_name]
+        else:
+            col_name = op.col_name
+        tmp_key, wrapper = "", ""
+        for col in col_name:
+            res = op(data[col])
+            assert isinstance(res, dict)
+            for key, val in res.items():
+                # Check if it has 1D features. If yes, convert to 2D features
+                if len(val.shape) == 1:
+                    if isinstance(val, ExtMemArrayWrapper):
+                        val = val.to_numpy().reshape(-1, 1)
+                    else:
+                        val = val.reshape(-1, 1)
+                if len(col_name) == 1:
```

Comment:
What if res.items() returns multiple key, val pairs? Are we going to only store the first one?

Reply:
You did not resolve my question.

Reply:
If we only expect res to return 1 (key, val). Please add an assert here and also a TODO.